src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -548,7 +542,7 @@ public static JsonTypeInfo<T> CreateJsonTypeInfo<T>(JsonSerializerOptions option
                 ThrowHelper.ThrowArgumentNullException(nameof(options));
             }
 
-            DefaultJsonTypeInfoResolver.RootDefaultInstance();
+            options.InitializeForReflectionSerializer();
```

Comment:
Hmmmm, again this seems to be opting in default contract resolution fallback which would be surprising behavior for somebody simply creating a blank JsonTypeInfo. Why is this needed?